WireAPI/Tests/WireAPITests/Network/NetworkService/NetworkServiceTests.swift

```diff
@@ -0,0 +1,109 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireTestingPackage
+import XCTest
+
+@testable import WireAPI
+@testable import WireAPISupport
+
+final class NetworkServiceTests: XCTestCase {
+
+    var sut: NetworkService!
+    var backendURL: URL!
+
+    override func setUp() async throws {
+        try await super.setUp()
+        backendURL = try XCTUnwrap(URL(string: "https://www.example.com"))
+        sut = NetworkService(baseURL: backendURL)
+        sut.configure(with: .mockURLSession())
+    }
+
+    override func tearDown() async throws {
+        backendURL = nil
+        sut = nil
+        try await super.tearDown()
+    }
+
+    // MARK: - Execute request
+
+    func testItDoesNotExecuteAnInvalidRequest() async throws {
```

Comment:
Not a request for a change but just so you know I started applying this naming convention [here](https://wearezeta.atlassian.net/wiki/spaces/IOS/pages/1434124328/Conventions) in many of my PRs (even if it's just a proposal for now). Maybe we could align on this, if so we should rename the test  : `testExecuteRequest_It_Throws_An_Invalid_Request_Error`

Reply:
agree let's find one and stick to it, no strong opinion on it

Reply:
ah yep, I forgot already :)

Reply:
done 59fc0edfab22f8e279a0ca8d305a48a501773914